### PR TITLE
Catalog platformA elements

### PR DESCRIPTION
**Pull request by tannoa2**
Catalog VPD, processors. dimms, and drives sub elements

### CODE REVIEW

File: lib/jobs/emc-redfish-catalog.js [hunk 1]
```diff
@@ -30,6 +31,8 @@ function EmcRedfishCatalogJobFactory(
 ) {
     var logger = Logger.initialize(EmcRedfishCatalogJobFactory);
 
+    var redfish = new RedfishTool();
```
Comment (jlongever) [inline lib/jobs/emc-redfish-catalog.js]: Could you move this in the constructor ?

**Comment by tannoa2** [comment]
@RackHD/corecommitters @jlongever @zyoung51 @keedya @uppalk1 @brianparry

File: lib/jobs/emc-redfish-catalog.js [hunk 2]
```diff
@@ -90,30 +91,81 @@ function EmcRedfishCatalogJobFactory(
...
+            assert.object(element.body);
+            return element.body;
+        })
+        .map(function(element){
+            _.forEach(element,function(value,key){
+                if(key === "Processors" || key === "Vpd" || key === "Dimms" || key === "Drives") {
+                    element = Promise.resolve(element)
+                        .then(function(element){
+                           return  self.updatingComponents(element,key);
+                        });
```
Comment (yyscamper) [inline lib/jobs/emc-redfish-catalog.js]: The `Promise.resolve(element)` is abundant here, let `element = self.updatingComponents(element,key)` is OK.

**Comment by yyscamper** [comment]
Still :+1:

**Comment by uppalk1** [comment]
:+1:

**Comment by jlongever** [comment]
+1

**Pull request merged**